src/renderers/shared/stack/reconciler/ReactInstanceType.js

```diff
@@ -0,0 +1,37 @@
+ /**
+ * Copyright 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ * @providesModule ReactInstanceType
+ */
+
+'use strict';
+
+export type DebugID = number;
+
+export type ReactInstance = {
```

Comment:
It took me a while to figure out how `instantiateReactComponent` worked, this is super convoluted :)